pyiceberg/io/pyarrow.py

```diff
@@ -1089,7 +1091,7 @@ def project_table(
             deletes_per_file.get(task.file.file_path),
             case_sensitive,
             limit,
-            table.name_mapping(),
+            None,
```

Comment:
I think we need to move `name_mapping()` method to table_metadata and call it here.
```python
table_metadata.name_mapping()
```
Otherwise, we no longer use the name-mapping feature during reading.

For example, if I addd
```
print(tbl.scan().to_pandas())
```
to the end of test: https://github.com/apache/iceberg-python/blob/main/tests/integration/test_add_files.py#L214
`project_table` will complain that field-id is not found in the added parquet file, but the table does have name-mapping configured.

Reply:
I opened a PR in your repo containing the change I proposed and the test that can re-produce the issue: https://github.com/Fokko/iceberg-python/pull/1

Reply:
Ah, yes, I just added `None` temporarily to silence the IDEA, but then forgot about it. Thanks for raising the PR :D Good to have this tested as well 👍